sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp

```diff
@@ -284,15 +285,25 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
   Azure::Core::Response<Models::DeleteFileResult> FileClient::Delete(
       const FileDeleteOptions& options) const
   {
-    Details::DataLakeRestClient::Path::DeleteOptions protocolLayerOptions;
-    protocolLayerOptions.LeaseIdOptional = options.AccessConditions.LeaseId;
-    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
-    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
-    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
-    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
-    auto result = Details::DataLakeRestClient::Path::Delete(
-        m_dfsUri, *m_pipeline, options.Context, protocolLayerOptions);
+    DeletePathOptions deleteOptions;
+    deleteOptions.AccessConditions = options.AccessConditions;
+    deleteOptions.Context = options.Context;
+    auto result = PathClient::Delete(deleteOptions);
+    auto ret = Models::DeleteFileResult();
+    ret.Deleted = true;
```

Comment:
Isn't this `true` already?

Reply:
Yes, but not relying on the default initialized value is a good practice as far as I'm concerned.

Reply:
Well, I don't agree but fine.